SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -711,17 +711,20 @@ - (void)sdl_onAppStateUpdated:(NSNotification *)notification {
 
 /// Handles a notification that the background task has ended. If the app is still in the background, the TCP transport disconnects. If the app has re-entered the foreground or the manager has shutdown then the notification is ignored.
 /// @return A background task ended handler
-- (nullable void (^)(void))sdl_backgroundTaskEndedHandler {
+- (nullable BOOL (^)(void))sdl_backgroundTaskEndedHandler {
     __weak typeof(self) weakSelf = self;
     return ^{
         __strong typeof(self) strongSelf = weakSelf;
-        if (strongSelf.sdl_getAppState == UIApplicationStateActive) {
-            SDLLogV(@"App has been foregrounded. Ignoring notification that the background task ended.");
+        if (strongSelf.sdl_getAppState == UIApplicationStateActive || [strongSelf.stateMachine isCurrentState:SDLSecondaryTransportStateStopped]) {
+            SDLLogV(@"No cleanup needed since app has been foregrounded.");
+            return NO;
         } else if ([strongSelf.stateMachine isCurrentState:SDLSecondaryTransportStateStopped]) {
-            SDLLogV(@"Manager has been stopped. Ignoring notification that the background task ended.");
+            SDLLogV(@"No cleanup needed since manager has been stopped.");
+            return NO;
         } else {
-            SDLLogD(@"Disconnecting TCP transport due to the background task ending.");
+            SDLLogD(@"Performing cleanup due to the background task expiring: disconnecting the TCP transport.");
```

Comment:
```suggestion
            // `endBackgroundTask` will be called when the secondary transport disconnects.
            SDLLogD(@"Performing cleanup due to the background task expiring: disconnecting the TCP transport.");
```

Reply:
I'm trying to trace how `disconnectSecondaryTransport` will be called through here. Somehow it seems to be that the TCP transport is cut off, which causes the protocol to be ended, which causes the audio / video service to end, which updates the `SDLStreamingMediaManager`. Can you help me understand how, from here, `disconnectSecondaryTransport` will be called by adding a comment here? I'm just envisioning a but in the secondary transport in the future and the pain caused by having to track this down without an over-abundance of documentation since this is so complicated.

Reply:
@NicoleYarroch This needs internal comment documentation. It's very confusing to try to trace even now. Trying to figure out what's going on a year from now would be very difficult. Please update comment documentation to explain what is happening here in detail.